pdns/recursordist/negcache.cc

```diff
@@ -97,8 +97,21 @@ bool NegCache::get(const DNSName& qname, const QType& qtype, const struct timeva
  *
  * \param ne The NegCacheEntry to add to the cache
  */
-void NegCache::add(const NegCacheEntry& ne) {
-  replacing_insert(d_negcache, ne);
+void NegCache::add(NegCacheEntry& ne) {
+  ne.d_bytes = sizeof(ne);
+  for (const auto& rec : ne.DNSSECRecords.records) {
+    ne.d_bytes+=rec.d_content->d_size_in_bytes;
+  }
+  for (const auto& sig : ne.DNSSECRecords.signatures) {
+    ne.d_bytes+=sig.d_content->d_size_in_bytes;
+  }
+
```

Comment:
We also need to account for the records and signatures in `authoritySOA`.

Reply:
ah, I missed those because I stole the code from `NegCache::dumpToFile` which also ignores i. I'll fix both.

Reply:
pushed